SAMI subs: start a new line if there's no space left to append to the
previous one.

Modified:
   trunk/sub/subreader.c

Modified: trunk/sub/subreader.c
==============================================================================
Index: mplayer/sub/subreader.c
===================================================================
--- mplayer.orig/sub/subreader.c
+++ mplayer/sub/subreader.c
@@ -172,6 +172,8 @@ static subtitle *sub_read_line_sami(stre
 	    break;
 
 	case 3: /* get all text until '<' appears */
+        if (p - text >= LINE_LEN)
+            sami_add_line(current, text, &p);
 	    if (*s == '\0') break;
 	    else if (!strncasecmp (s, "<br>", 4)) {
                 sami_add_line(current, text, &p);
